### ISSUE TITLE
non-graded (0/0 pts) assignments should have a nicer UI

### ISSUE THREAD

**Issue by jaguarfi** [apluslms/a-plus#488]
The current UX is both confusing in various ways and even factually incorrect (e.g. submission shows up as green ("correct") before even being assessed). How these assignments should up in different places (in-chapter; in feedback dialogue; on student's status page) needs a rethink.

**Comment by raphendyr** [comment]
Small points could (or should) have textual values:

* No points, but a submission
  * 0 / 0 -> "ok" or "completed"
* Two values, i.e., max 1 point
  * 0 / 1 -> "failed"
  * 1 / 1 -> "accepted", "success" or "ok"
* Three values, i.e, max 2 points
  * 0 / 2 -> "failed"
  * 1 / 2 -> "ok"
  * 2 / 2 -> "good"

**Comment by etanttila** [comment]
> 
>     * Three values, i.e, max 2 points
>       
>       * 0 / 2 -> "failed"
>       * 1 / 2 -> "ok"
>       * 2 / 2 -> "good"

Why are exercises with max 2 points any different than ones with say 3 or 5? I feel like often 1/2 means that there are two different questions in the same exercise or two parts to the problem and one of them is done correctly and the other isn't, which means it isn't ok but perhaps halfway done.

I definitely think that 0/0 needs to be rethinked, but I don't really see the need for textual values for 1- and 2-point exercises.

**Comment by Mankro** [comment]
I don't think that we could always assume that 1-point or 2-point exercises should use those special failed/accepted statuses. There are exercises that are used as forms to collect to information and the 1-2 points are used to signal that the data was stored, while sometimes there are real exercises that are worth a small number of points. Adding these special failed/accepted statuses to normal exercises would be confusing and misleading. The max number of points alone does not imply that the exercise would not be a normal graded exercise.

**Comment by raphendyr** [comment]
Personally, I would consider all exercises with less than 10 points as special cases. However, it's true that someone could use small numbers for to calculate points. Though, it could be possible to make this rule, assuming there is sufficient research stating that bigger numbers work better to motivate people.

Of course, from implementation point of view, the A+ config could include variable which states if max_points <= 2 should use textual format or not (0/0 would always be textual). The aplus-rst-tools could set this config to true by default, but would allow changing it of course (i.e. by default aplus-rst-tools course would use textual representation).

**Comment by Mankro** [comment]
We really can't assume that all exercises with less than 10 points are special cases. Some courses use small numbers of points and it is the teacher's choice. We can't force people to use large numbers for points just because students might be motivated by big numbers. The teachers decides the grading scale and the number of points and the system must not assume special cases just due to a smaller number of points.

**Comment by raphendyr** [comment]
As an additional benefit,  a points type information would help with grade calculation. If the points are textual, then those can not be summed or averaged, however those can be used as a boolean to indicat,e if the exercise has been completed.
